### DISCUSSION TITLE
Alternative port for NTP?

### DISCUSSION THREAD

**Discussion by pkkrusty** [arendst/Tasmota#11147] [Q&A] [answered]
After searching the documentation, I couldn't find anything about whether it's possible to point the NTPServer command to a port other than 123. Anyone know? If so, what's the syntax?

Reason: I have an Android phone that I'm syncing with GPS time, and using it as a NTP server, but the version of Android it's running won't allow port 123 to be used. My low-cost fix until I get a Raspberry Pi in a few months to run all my tasmota-related services.

**Comment by sfromis** [comment, marked as answer]
NTP port number 123 is [hardcoded](https://github.com/arendst/Tasmota/blob/development/tasmota/support_wifi.ino#L781-L794) in the source. You can change and recompile if you want.

**Comment by Jason2866** [comment]
Or you add a GPS receiver to a Tasmota device. Tasmota NTP time server. Cheap and reliable

**Reply by pkkrusty** [reply]
I thought about that.  I'll probably end up doing that in the long run, but the Android phone is a cheap(er) solution at the moment.  Unfortunately, I found out this morning that the phone I want to use suffers from the week rollover bug that started in April 2019, so can't be used as a time source, since it thinks it's 1024 weeks earlier than it actually is. No firmware updates are forthcoming, so I'm shelving the effort for now.

Even cheaper option is using a RTC connected to Tasmota, and counting on the internet to function at least every couple days to keep it accurate. I'm in rural Africa, and the less I have to rely on public services, the better.
